rasa/utils/endpoints.py

```diff
@@ -73,18 +73,19 @@ class EndpointConfig:
 
     def __init__(
         self,
-        url: Text = None,
-        params: Dict[Text, Any] = None,
-        headers: Dict[Text, Any] = None,
-        basic_auth: Dict[Text, Text] = None,
+        url: Optional[Text] = None,
+        params: Optional[Dict[Text, Any]] = None,
+        headers: Optional[Dict[Text, Any]] = None,
+        basic_auth: Optional[Dict[Text, Text]] = None,
         token: Optional[Text] = None,
         token_name: Text = "token",
         **kwargs: Any,
     ) -> None:
+        """Creates an `EndpointConfig` instance."""
         self.url = url
```

Comment:
should we immediately raise if this is `None`?

Reply:
Apparently there's code that depends on the current behavior. Things failed when I changed the default before. So, not sure it's worth digging to deep here rn 🤔